components/needs_team/needs_team.jsx

```diff
@@ -81,35 +84,104 @@ export default class NeedsTeam extends React.Component {
         this.state = {
             team,
             finishedFetchingChannels: false,
+            attemptingToJoinTeam: !team,
         };
+
+        if (!team) {
+            this.joinTeam(this.props);
+        }
     }
 
     UNSAFE_componentWillReceiveProps(nextProps) { // eslint-disable-line camelcase
         if (this.props.match.params.team !== nextProps.match.params.team) {
-            this.setState({team: this.updateCurrentTeam(nextProps)});
+            const team = this.updateCurrentTeam(nextProps);
+            this.setState({
+                team,
+                attemptingToJoinTeam: !team,
+            });
+            if (!team) {
+                this.joinTeam(nextProps);
+            }
         }
     }
 
-    updateCurrentTeam(props) {
-        // First check to make sure you're in the current team
-        // for the current url.
-        const teamName = props.match.params.team;
-        const team = TeamStore.getByName(teamName);
+    componentDidMount() {
+        startPeriodicStatusUpdates();
+        startPeriodicSync();
 
-        if (!team) {
-            props.history.push('/?redirect_to=' + encodeURIComponent(props.location.pathname));
-            return null;
+        // Set up tracking for whether the window is active
+        window.isActive = true;
```

Comment:
Do we need to set this to false on componentWillUnmount?

Reply:
Looks like it is better to set it to false as not to updates on socket emits.